pkg/autoupgrade/daemon.go

```diff
@@ -319,18 +319,20 @@ func AutoUpgradePattern(image string) (string, bool) {
 	return tag, strings.ContainsAny(tag, "#*")
 }
 
-// ImpliedAutoUpgrade returns boolean indicating if auto-upgrade is implied either
-// by a pattern in the app.Spec.image or app.Spec.AutoUpgradeInterval being specified.
-func ImpliedAutoUpgrade(interval, image string) bool {
-	if _, isPattern := AutoUpgradePattern(image); isPattern || interval != "" {
+// Implied returns a boolean indicating if auto-upgrade is implied by either:
+// 1. A pattern specified in the image
+// 2. A non-empty interval
+// 3. app.Spec.NotifyUpgrade set to true
+func Implied(image, interval string, notify bool) bool {
+	if _, isPattern := AutoUpgradePattern(image); isPattern || interval != "" || notify {
 		return true
 	}
 	return false
 }
 
 func Mode(appSpec v1.AppInstanceSpec) (string, bool) {
 	_, isPat := AutoUpgradePattern(appSpec.Image)
-	on := appSpec.GetAutoUpgrade() || appSpec.GetNotifyUpgrade() || isPat
+	on := appSpec.GetAutoUpgrade() || isPat
```

Comment:
Do we care about existing apps? If a user has an existing app that has only "notify upgrade" enabled, it seems that their apps will stop getting upgraded after this change. Is that correct? Is that what we want?

Reply:
That's a good question. In my mind this is intended since this is no longer behavior we want to support. However, I think it's worth raising to the team to be sure so I'll go ahead and ask.